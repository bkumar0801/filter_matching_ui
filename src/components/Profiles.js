import React from 'react';
import '../style/App.css';

class Profiles extends React.Component {
    render() {
      if(this.props.profiles.length !== 0) {
       return (
        <div className="feed">
          {this.props.profiles.map(c => <Profile displayName={c.displayName} job={c.job}
          age = {c.age} height = {c.height} photo = {c.photo} religion = {c.religion}
          city = {c.city} contact = {c.contact} favourite = {c.favourite} score = {c.score}/>)}
        </div>
      );
     }
     else {
       return (
         <div className="section-label">
           No result found!
         </div>
       );
     }
    }
  }
 
 class Profile extends React.Component {
   render() {
     return (
     <div className="profile">
       <img src={this.props.photo} className="img-circular" alt="profile-img"/>
       <div className="display-job">
         <span className="name">{this.props.displayName}</span>
         <span className="job-title">{this.props.job}</span>
       </div>
       <div className="extra-content">
         <span className="content">{this.props.age}</span>
         <span className="content">{this.props.height}</span>
       </div>
       <div className="extra-content">
         <span className="content">{this.props.religion}</span>
         <span className="content">{this.props.city}</span>
       </div>
       <div className="extra-content">
         <span className="content">{this.props.contact}</span>
         <span className="content">{this.props.favourite}</span>
       </div>
       <div className="extra-content">
         <span className="content">{this.props.score}</span>
       </div>
     </div>)
   }
  }

  export default Profiles;
 