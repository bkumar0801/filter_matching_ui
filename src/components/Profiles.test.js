import React from 'react';
import { shallow, enzyme } from 'enzyme';
import Profiles from './Profiles';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import {sinon, spy, stub} from 'sinon';

configure({ adapter: new Adapter() });

describe(Profiles, () => {
    const mockOnInput = jest.fn();
    const emptyProfiles = []
    const profiles = [{
        displayName: "Test",
        job: "Engineer",
        age: "Age: 18 yrs",
        height: "Height: 100 cm",
        photo: "test pic",
        religion: "Religion: Hindu",
        city: "City: Test",
        contact: "Contacts Exchanged: 2",
        favourite: "Favourite: True",
        score: "Compatibility Score: 90%"
      },
    ];
    const component = shallow(<Profiles profiles = {profiles} />); 

    it('renders and matches our snapshot with empty profile', () => {
        const component = renderer.create(<Profiles profiles = {emptyProfiles}/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders and matches our snapshot with one profile', () => {
        const component = renderer.create(<Profiles profiles = {profiles}/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});