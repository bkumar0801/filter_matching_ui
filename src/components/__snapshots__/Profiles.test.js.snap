// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class Profiles extends _react2.default.Component {
  render() {
    if (this.props.profiles.length !== 0) {
      return (
        _react2.default.createElement('div', { className: 'feed', __source: { fileName: _jsxFileName, lineNumber: 8 }, __self: this },
          this.props.profiles.map(c => _react2.default.createElement(Profile, { displayName: c.displayName, job: c.job,
            age: c.age, height: c.height, photo: c.photo, religion: c.religion,
            city: c.city, contact: c.contact, favourite: c.favourite, score: c.score, __source: { fileName: _jsxFileName, lineNumber: 9 }, __self: this }))));


    } else
    {
      return (
        _react2.default.createElement('div', { className: 'section-label', __source: { fileName: _jsxFileName, lineNumber: 17 }, __self: this }, 'No result found!'));



    }
  }} renders and matches our snapshot with empty profile 1`] = `
<div
  className="section-label"
>
  No result found!
</div>
`;

exports[`class Profiles extends _react2.default.Component {
  render() {
    if (this.props.profiles.length !== 0) {
      return (
        _react2.default.createElement('div', { className: 'feed', __source: { fileName: _jsxFileName, lineNumber: 8 }, __self: this },
          this.props.profiles.map(c => _react2.default.createElement(Profile, { displayName: c.displayName, job: c.job,
            age: c.age, height: c.height, photo: c.photo, religion: c.religion,
            city: c.city, contact: c.contact, favourite: c.favourite, score: c.score, __source: { fileName: _jsxFileName, lineNumber: 9 }, __self: this }))));


    } else
    {
      return (
        _react2.default.createElement('div', { className: 'section-label', __source: { fileName: _jsxFileName, lineNumber: 17 }, __self: this }, 'No result found!'));



    }
  }} renders and matches our snapshot with profile 1`] = `
<div
  className="feed"
>
  <div
    className="profile"
  >
    <img
      alt="profile-img"
      className="img-circular"
      src="test pic"
    />
    <div
      className="display-job"
    >
      <span
        className="name"
      >
        Test
      </span>
      <span
        className="job-title"
      >
        Engineer
      </span>
    </div>
    <div
      className="extra-content"
    >
      <span
        className="content"
      >
        Age: 18 yrs
      </span>
      <span
        className="content"
      >
        Height: 100 cm
      </span>
    </div>
    <div
      className="extra-content"
    >
      <span
        className="content"
      >
        Religion: Hindu
      </span>
      <span
        className="content"
      >
        City: Test
      </span>
    </div>
    <div
      className="extra-content"
    >
      <span
        className="content"
      >
        Contacts Exchanged: 2
      </span>
      <span
        className="content"
      >
        Favourite: True
      </span>
    </div>
    <div
      className="extra-content"
    >
      <span
        className="content"
      >
        Compatibility Score: 90%
      </span>
    </div>
  </div>
</div>
`;
